webviz_subsurface/containers/_intersect.py

```diff
@@ -0,0 +1,297 @@
+                                     for i in ['Name', 'TVDmin', 'TVDmean',
+                                               'TVDmax', 'TVDstddev']]
+                        )
+                    ])
+                ]),
+                html.Div(style={'height': '80vh'}, children=[
+                    dcc.Graph(style={'height': '80vh'}, id='graph'),
+                    html.Div(style=self.style_fence_options, children=[
+                        html.P('Start depth:', style={'font-weight': 'bold'}),
+                        html.P('Graph zoom level:', style={
+                               'font-weight': 'bold'}),
+                        dcc.Input(style={'overflow': 'auto'},
+                                  id='well-tvd', type='number', value=0),
+                        dcc.RadioItems(
+                            id='ui_revision',
+                            options=[{'label': k, 'value': v}
+                                     for k, v in {
+                                     'Keep on new data': True,
+                                     'Reset on new data': False}.items()],
+                            value=True,
+                        ),
+                    ])
+
+                ])
+            ])
+        ])
+
+    def set_callbacks(self, app):
+        @app.callback(
+            Output('graph', 'figure'),
+            [Input('well_list', 'value'),
+             Input('real_list', 'value'),
+             Input('surf_list', 'value'),
+             Input('well-tvd', 'value'),
+             Input('ui_revision', 'value')])
+        def set_fence(_well_path, _reals, _surfs, _tvdmin, _keep_zoom_state):
+            if not isinstance(_surfs, list):
+                _surfs = [_surfs]
+            if not isinstance(_reals, list):
+                _reals = [_reals]
+            if _reals == []:
+                _reals = list(self.realizations.keys())
+            if _surfs == []:
+                _surfs = self.surface_names
+            s_names = [s for s in self.surface_names if s in _surfs]
+            xsect = self.plot_xsection(_well_path, _reals, s_names, _tvdmin)
+            layout = self.graph_layout
+            if _keep_zoom_state:
+                layout['uirevision'] = 'keep'
+            return {'data': xsect, 'layout': layout}
+
+        @app.callback(Output('table', 'data'),
+                      [Input('graph', 'hoverData')],
+                      [State('graph', 'figure'),
+                       State('surf_list', 'value')])
+        def hover(_data, _fig, _surfaces):
+            try:
+                graph = _fig['data']
+            except TypeError:
+                return [{
+                    'TVDmin': None,
+                    'TVDmean': None,
+                    'TVDmax': None,
+                    'TVDstddev': None,
+                    'Name': None
+                }]
+            if not isinstance(_surfaces, list):
+                _surfaces = [_surfaces]
+            if _surfaces == []:
```

Comment:
[`if not _surfaces:`](https://stackoverflow.com/questions/53513/how-do-i-check-if-a-list-is-empty)